From 74e0f5863f187fa61587f9dcecb9a9adab56add9 Mon Sep 17 00:00:00 2001
From: Alessandro Astone <alessandro.astone@canonical.com>
Date: Wed, 26 Mar 2025 17:50:08 +0100
Subject: [PATCH] atspi-device-a11y-manager: Fix crash on dispose

grabbed_modifiers is a simple list of integers, we cannot call g_free on the
list items. Fixes SIGSEGV on disposal, as seen in Orca and Mutter 48.0
---
 atspi/atspi-device-a11y-manager.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/atspi/atspi-device-a11y-manager.c b/atspi/atspi-device-a11y-manager.c
index 92365d35..1939b3bf 100644
--- a/atspi/atspi-device-a11y-manager.c
+++ b/atspi/atspi-device-a11y-manager.c
@@ -386,7 +386,7 @@ atspi_device_a11y_manager_dispose (GObject *object)
 {
   AtspiDeviceA11yManager *device = ATSPI_DEVICE_A11Y_MANAGER (object);
 
-  g_slist_free_full (device->grabbed_modifiers, g_free);
+  g_slist_free (device->grabbed_modifiers);
   g_slist_free_full (device->grabbed_keys, g_free);
   g_slist_free_full (device->virtual_modifiers, g_free);
   if (device->refresh_timeout_id)
-- 
GitLab

